### ISSUE TITLE
Classifier loss misunderstand

### ISSUE THREAD

**Issue by karenyun** [matterport/Mask_RCNN#686]
Hi, the loss of mrcnn_class seems to be different from the original paper.
`# Find predictions of classes that are not in the dataset.
    pred_class_ids = tf.argmax(pred_class_logits, axis=2)

    # TODO: Right now it assumes all images in a batch have the same active_class_ids
    pred_active = tf.gather(active_class_ids[0], pred_class_ids)
# Loss
    loss = tf.nn.sparse_softmax_cross_entropy_with_logits(
        labels=target_class_ids, logits=pred_class_logits)

    # Erase losses of predictions of classes that are not in the active classes of the image.
    loss = loss * pred_active` 

If the total dataset has 80 classes which means the classifier will not predict the classes outside the 80, so in this way, the loss only includes the values of pred_classes rather than 80 values. But in the fast-rcnn, the classifier loss does not define as it. And I do not understand the advantages of doing in this way. Thanks!
